### PR TITLE
feat(language): allow tweaking version cache

### PR DESCRIPTION
**Pull request by JanDeDobbeleer**
### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md].
- [x] The commit message follows the [conventional commits][cc] guidelines.
- [x] Tests for the changes have been added (for bug fixes / features).
- [x] Docs have been added/updated (for bug fixes / features).

[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md
[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: website/docs/segments/languages/zig.mdx [hunk 1]
```diff
@@ -19,7 +19,8 @@ import Config from "@site/src/components/Config.js";
     powerline_symbol: "\ue0b0",
     foreground: "#342311",
     background: "#ffad55",
-    template: " \ue6a9 {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }} ",
+    template:
+      " \ue6a9 {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }} ",
```
Comment (lewis-yeung) [inline website/docs/segments/languages/zig.mdx:23]: @JanDeDobbeleer Is this newline added by mistake?

Comment (JanDeDobbeleer) [reply website/docs/segments/languages/zig.mdx:23]: Seems so, thanks!

Comment (JanDeDobbeleer) [reply website/docs/segments/languages/zig.mdx:23]: ah no, it's the formatter that does this.

**Comment by JanDeDobbeleer** [comment]
@lewis-yeung `.Text` can never be taken into account for cross template properties for primary and right, only for extra prompts. The reason is that the execution happens in parallel and writing sequentially. The check looks at if the segment has been executed in case there's a need in a template, but `.Text` is only available after writing, and thus after that check happens, as needs is a precondition for resolving `.Text`. Theoretically a check could be built-in that validates the presence of `.Text` and its value, but that would only work sequentially and in my opinion, that logic isn't worth my while as I don't really see the use case in primary and right.

**Comment by lewis-yeung** [comment]
@JanDeDobbeleer Got it. I was taking that as an example to reproduce the issue, but not using it in my own config.

**Comment by JanDeDobbeleer** [comment]
@lewis-yeung It was a really good catch, but that should be the only case where this behavior occurs, which I can live with as "by design". I want to keep that part as simple as possible. If you notice any other behavior, let me know!

**Comment by lewis-yeung** [comment]
@JanDeDobbeleer Unfortunately, reverting the fix brought back the issue of #2885. It occurs not only when cross-referencing the `.Text` of a segment, but **any property**.

**Comment by JanDeDobbeleer** [comment]
@lewis-yeung I made sure to test this with cross segment properties. I can double check, but this shouldn't be the case though.

**Comment by lewis-yeung** [comment]
@JanDeDobbeleer It should work in an extra prompt, right? However, I always see the issue when using a cross segment property in the transient prompt, but not with the previous fix of ce027593b1bc397b0b9c65749c0020c900466c00.

**Comment by JanDeDobbeleer** [comment]
@lewis-yeung ah, that's a good one. I didn't explicitly check that. It definitely works in primary and right as otherwise #5813 would be broken again. That's the one that broke when fixing #2885 previously.

**Comment by JanDeDobbeleer** [comment]
@lewis-yeung I can't reproduce this though, my transient prompt references the `git` segment successfully (text and properties).

**Comment by lewis-yeung** [comment]
@JanDeDobbeleer Have you checked on Windows? It happens more often on Windows.

**Comment by JanDeDobbeleer** [comment]
@lewis-yeung I only test on Windows. Can't reproduce it at all.

**Comment by JanDeDobbeleer** [comment]
@lewis-yeung care to share the config?

**Comment by lewis-yeung** [comment]
A minimal example:

```yaml
version: 3
patch_pwsh_bleed: true
final_space: true
blocks:
  - type: prompt
    alignment: left
    segments:
      - type: os
        style: plain
        properties:
          windows: ' '
        template: '{{.Icon}}'
      - type: text
        style: plain
        template: foo|
      - type: text
        style: plain
        template: bar|
      - type: text
        style: plain
        template: baz|
transient_prompt:
  template: '->{{.Segments.Os.Icon}}<-'
```

Result:

screenshot

**Comment by JanDeDobbeleer** [comment]
@lewis-yeung I'll test after dinner.  Using this extensively on my setup as well, haven't noticed that.

**Comment by JanDeDobbeleer** [comment]
@lewis-yeung I can reproduce this, but it seems to be linked to extra only, non-extra isn't impacted.

**Comment by JanDeDobbeleer** [comment]
@lewis-yeung so, the issue is with the key value map that's behind the concurrent map. Everything gets added, but not always correctly. Sometimes there seems to be issues in concurrency, even when that should not happen at all. And when it happens, the cache writes an empty map, my assumption is that the memory layout somehow breaks.

**Pull request merged**